crates/turbopack/src/module_options/mod.rs

```diff
@@ -249,19 +256,44 @@ impl ModuleOptionsVc {
                     } else {
                         None
                     },
-                    Some(ModuleRuleEffect::ModuleType(ModuleType::Css(
-                        css_transforms,
-                    ))),
+                    Some(ModuleRuleEffect::ModuleType(ModuleType::CssGlobal)),
                 ]
                 .into_iter()
                 .flatten()
                 .collect(),
             ),
             ModuleRule::new(
-                ModuleRuleCondition::ResourcePathEndsWith(".module.css".to_string()),
-                vec![ModuleRuleEffect::ModuleType(ModuleType::CssModule(
-                    css_transforms,
-                ))],
+                ModuleRuleCondition::all(vec![
+                    ModuleRuleCondition::ResourcePathEndsWith(".module.css".to_string()),
+                    ModuleRuleCondition::not(ModuleRuleCondition::ReferenceType(
```

Comment:
This is pretty weird. We should not confront the user with internal modules by default.

Rules shouldn't match internal references by default. Rule should need to opt-in to matching internal references.

To do that, add `ReferenceType::is_internal()` that returns true for all `Internal` references.

Also add `ModuleRuleCondition::Internal`. When `is_internal()` returns true, a rule will only match if `ModuleRuleCondition::Internal` is specified.

This way it doesn't match rules for internal references by default, only when opt-in to that. That also applies to EcmascriptReferenceType::Internal.